### PR TITLE
Fix syntax highlighting for multiline lang/type attributes in style and script tags

### LINKED ISSUE #14657: 🐛 BUG: SASS content incorrectly highlighted as CSS syntax when `lang='sass'` is not on the same line with `<style`
### Describe the Bug

When the `lang` attribute of the `<style>` tag is not on the same line as the `style` tag, the content may be incorrectly highlighted as CSS syntax.

### Steps to Reproduce

1. `npm create astro@latest`
2. open the project in vscode
3. install newest astro extension
4. create `anyname.astro` with the contents
```astro
<style
  lang="sass"
  define:vars={{
    // any comment to avoid formatter collapse content into a single line
    foo: '#fff',
  }}
>
  .bar
    color: var(--foo)
</style>
```
5. the content below `.bar` (include `</style>`) will be incorrectly highlighted as css syntax, not sass.

### Link to Minimal Reproducible Example

https://gist.github.com/HPCesia/6e935433cfd6e414099f71714cb4961d

### PR DESCRIPTION
**Pull request by svssdeva**
## Changes

- Style and script tags with `lang` or `type` on a different line than the tag name now get correct syntax highlighting. Previously, the TextMate grammar's `\G` anchor only matched on the first line after the opening tag, causing multiline attributes to fall through to the default language (CSS/JS) instead of the specified one (Sass, SCSS, TypeScript, etc.)
- Adds a `begin/while` pattern that consumes intermediate attribute lines until the line containing `lang`/`type` is reached, then hands off to the existing lang-detection rules without the `\G` constraint
- Includes explicit rules for `application/ld+json` → JSON and `type="module"` → JavaScript to handle cases where backreference capture groups reset after the `while` rule

Closes #14657

## Testing

- Added 4 grammar snapshot fixtures covering multiline `lang` detection: `sass-multiline.astro`, `scss-multiline.astro`, `ts-multiline.astro`, and `sass-multiline-scoped.astro` (three-line split with `is:global`)
- All 26 grammar tests pass, including existing single-line fixtures confirming no regressions

## Docs

- No docs update needed — this is a bug fix in the VS Code extension's TextMate grammar with no user-facing API change

### CODE REVIEW

File: packages/language-tools/vscode/syntaxes/astro.tmLanguage.src.yaml [hunk 1]
```diff
@@ -519,6 +519,24 @@ repository:
         end: (?=</|/>)
         name: meta.lang.$3.astro
         patterns: [include: '#tags-lang-start-attributes']
+      # Multiline: consume attribute lines until lang/type or > is encountered.
+      - begin: \G(?=[^>]*$)
+        while: ^(?!.*?(type|lang)\s*=)(?![^>]*>)
```
Comment (mazze93) [inline packages/language-tools/vscode/syntaxes/astro.tmLanguage.src.yaml:524]: Have you tested this `while` condition against a malformed/unclosed tag (e.g. `<style` with no closing `>` before EOF)? The negative lookahead pair `(?!.*?(type|lang)\s*=)(?![^>]*>)` will keep the while-loop consuming lines until one of those conditions flips true or the file ends. Worth adding a fixture for that edge case to confirm it terminates gracefully rather than swallowing the rest of the document.

File: packages/language-tools/vscode/syntaxes/astro.tmLanguage.src.yaml [hunk 2]
```diff
@@ -519,6 +519,24 @@ repository:
...
+      # Multiline lang detection: fires on the line where lang/type appears (after while rule ends).
+      - begin: (?=\s*[^>]*?(type|lang)\s*=\s*(['"]|)(?:text\/)?(application\/ld\+json)\2)
+        end: (?=</|/>)
+        name: meta.lang.json.astro
+        patterns: [include: '#tags-lang-start-attributes']
+      - begin: (?=\s*[^>]*?(type|lang)\s*=\s*(['"]|)(module)\2)
+        end: (?=</|/>)
+        name: meta.lang.javascript.astro
+        patterns: [include: '#tags-lang-start-attributes']
+      - begin: (?=\s*[^>]*?(type|lang)\s*=\s*(['"]|)(?:text/|application/)?([\w\/+]+)\2)
```
Comment (mazze93) [inline packages/language-tools/vscode/syntaxes/astro.tmLanguage.src.yaml:536]: Nice generic fallback with `([\w/+]+)` → `meta.lang.$3.astro` — avoids hardcoding every possible lang value, which is more maintainable than enumerating languages one by one. One question: for an unrecognized/garbage `lang` value (e.g. `lang="nonsense"`), does this just produce an unmatched `meta.lang.nonsense.astro` scope that TextMate silently no-ops on, or could it visually break highlighting? A fixture covering an invalid lang value would make this bulletproof.

**Review by mazze93 — COMMENTED** [review]
Reviewed the grammar diff in astro.tmLanguage.src.yaml. My honest take: this is the better of the two open PRs fixing #14657.

The core mechanism is right: \G(?=[^>]*$) + while: ^(?!.*?(type|lang)\s*=)(?![^>]*>) correctly solves the root cause -- the original grammar's \G anchor was implicitly line-bound, so it silently gave up scanning for lang/type the moment the attribute wrapped to a new line. This fix keeps consuming lines until either the attribute or the closing > shows up, which is the actual fix rather than a workaround.

What I like specifically vs. the alternative PR (#16991): the generic capture rule ([\w/+]+) -> meta.lang.$3.astro means new language values don't require touching the grammar again, and the explicit special-casing of application/ld+json and module ahead of the generic fallback is the correct order of operations since those need distinct scope names. Fixture coverage here is also broader (ts, scss, sass, less, json, module all covered), and all 25 CI checks including the dedicated language-tools grammar suite are green.

Two things I'd want closed out before merge (left as inline comments): a fixture for a malformed/unclosed tag to confirm the while-loop terminates cleanly rather than consuming to EOF, and a fixture for an unrecognized lang value to confirm it degrades gracefully instead of producing a broken scope.

Opinion: maintainers should converge on this PR over #16991 rather than merging both -- they solve the same problem with a near-identical technique, but this one has the more maintainable generic capture and wider test coverage. Would rather see the two edge cases above added here than see a third competing PR opened.

**Comment by ematipico** [comment]
That's been fixed. Thank you
